### PR TITLE
docs: write call-existing-api.md

### LINKED ISSUE #694: Document the different backend use cases
Write documentation about how to consume backing services from Backstage.

Outline the difference between (1) just frontend plugin talking directly to a service, (2) frontend talking via the Backstage proxy to a service, (3) frontend with accompanying Backstage backend part. Give guidance on when to choose which one.

### PR DESCRIPTION
**Pull request by freben**
#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Comment by freben** [comment]
Fixes #694

File: docs/plugins/call-existing-api.md [hunk 1]
```diff
@@ -1,6 +1,168 @@
...
 
-## TODO
+This article describes the various options that Backstage frontend plugins have,
+in communicating with service APIs that already exist. Each section below
+describes a possible choice, and the circumstances under which it fits.
+
+## Issuing Requests Directly
+
+The most basic choice available is to issue requests directly from the plugin
+frontend code to the API, using for example `fetch` or a support library such as
```
Comment (stefanalund) [inline docs/plugins/call-existing-api.md:13]: Maybe clarify here that “the API” is whatever API and not part of Backstage.

File: docs/plugins/call-existing-api.md [hunk 2]
```diff
@@ -1,6 +1,168 @@
...
+choice for plugin makers. Since we have DNS based service discovery in place and
+a microservices framework that made it trivial to expose plain HTTP, it has been
+a matter of just adding a few lines of Backstage config to get the benefit of
+being easily and robustly reachable from users' web browsers as well.
+
+This may be used instead of direct requests, when:
+
+- You need to perform HTTPS termination and/or CORS handling, because the API
+  itself is not supplying those.
+- You need to inject a simple static secret into the requests, e.g. an
```
Comment (stefanalund) [inline docs/plugins/call-existing-api.md:100]: How do I do that? Link to docs that describe it perhaps.

File: docs/plugins/call-existing-api.md [hunk 3]
```diff
@@ -1,6 +1,168 @@
...
+that want to resolve a lot of sparse data from a larger list.
+
+This may be used instead of the above, when:
+
+- You need to perform complex model conversion, or protocol translation beyond
+  what the proxy handles.
+- You want to perform aggregations or summaries on the backend instead of on the
+  frontend.
+- You want to enable batching or caching of slower or more unreliable APIs.
+- You need to maintain state for your plugin, perhaps using the builtin database
```
Comment (stefanalund) [inline docs/plugins/call-existing-api.md:153]: Docs?

File: docs/plugins/call-existing-api.md [hunk 4]
```diff
@@ -1,6 +1,168 @@
...
+  ingress with just the Backstage config.
+
+## Creating a Backstage Backend Plugin
+
+Much like the Backstage frontend, the Backstage backend also has a plugin
+system. The above mentioned proxy is actually one such plugin. When you are in
+need of a more involved integration than just direct access to an API, or if you
+need to hold state, you may want to make such a plugin.
+
+Example:
```
Comment (stefanalund) [inline docs/plugins/call-existing-api.md:115]: Can we also link to an example plugin?

**Review by stefanalund — APPROVED** [review]

File: docs/plugins/call-existing-api.md (hunk 2 above)
Comment (freben) [reply docs/plugins/call-existing-api.md:100]: Adding headers is described in the http-proxy-middleware docs linked above

File: docs/plugins/call-existing-api.md (hunk 3 above)
Comment (freben) [reply docs/plugins/call-existing-api.md:153]: Linked to lighthouse

**Review by stefanalund — APPROVED** [review]

**Pull request merged**